service/history/workflow/mutable_state_impl.go

```diff
@@ -2639,6 +2640,12 @@ func (ms *MutableStateImpl) UpdateBuildIdAssignment(buildId string) error {
 	return ms.updateBuildIdsSearchAttribute(&commonpb.WorkerVersionStamp{UseVersioning: true, BuildId: buildId}, limit)
 }
 
+// For v3 versioned workflows (ms.GetEffectiveVersioningBehavior() != UNSPECIFIED), this will append a tag formed as
+// reachability:<effective_behavior>:<deployment_series_name>:<deployment_build_id> to the BuildIds search attribute,
```

Comment:
hm.. I think the reason to get reachability of unpinned would be to be able to further segment the "reachable" value into "reachable by unpinned only", which would mean you can feel more comfortable decommissioning a deployment, if it is only "reachable by unpinned"

Reply:
but now, we just treat "reachable by unpinned only" as equivalent to "unreachable"

Reply:
This plus my previous comment makes me thing maybe instead of a general `reachability:...` value, we need a `pinned:<deployment_series_name>:<deployment_build_id>` only for pinned wfs.

(later we can add `incompatible:<deployment_series_name>:<deployment_build_id>` for failed transitions)

Reply:
yeah you're right, that was our reason. But we abandoned that thinking. but why spend costly visibility space for something we don't need? If we want to add it in the future, I think we can start adding this SA to unpinned as well (as `unpinned:...`)

Reply:
ok! I agree. This is good. Glad we caught it -- hard to keep up with all the fast changes

Reply:
wait actually, I think we want `<behavior>:<deployment_series_name>:<deployment_build_id>`, because when a deployment changes from pinned to unpinned, we need to update the SA to not say "pinned" anymore

Reply:
or, to save storage, we can remove the previous pinned SA (I guess I can just remove the string from the string list that I read, and then write the new list?)

Reply:
yeah should at most be one pinned:... value in the list. Even if we keep reachability:... there should at most be one of those values in the list.